src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
```

Comment:
Just a note, we are using `@@redux-undo/INIT` here, **not** `@@redux/INIT`. We can deal with our own init action in any way we please, just don't touch `@@redux/INIT` :wink: